src/pybind/mgr/mgr_module.py

```diff
@@ -920,6 +940,184 @@ def cluster_log(self, channel: str, priority: ClusterLogPrio, message: str) -> N
     def version(self) -> str:
         return self._version
 
+    def pool_exists(self, name: str) -> bool:
+        pools = [p['pool_name'] for p in self.get('osd_map')['pools']]
```

Comment:
maybe a future todo, copying/jsonifying the osdmap on every access to self.db could get expensive for large clusters. it'd be better to use a get_pools() or pool_exists() c++ method - could be added to mgr/PyOSDMap.cc so you could do self.get_osdmap().get_pools() or something similar

Reply:
This only happens if `self._db is None` (usually the first access by each module).